tests/providers/gcp/gcp_network_test.py

```diff
@@ -0,0 +1,507 @@
+        self.assertDictContainsSubset({'cidr': '192.168.1.0/24'}, spec.custom_subnets['vm_2'])
+        self.assertLen(spec.networks, 2)
+        for k in spec.networks.keys():
+            self.assertItemsEqual(['1.2.3.4/33', '192.168.1.0/24'], spec.networks[k].all_nets)
+
+    def testLoadSameZoneCidrConfig(self):
+        spec = self._CreateBenchmarkSpecFromYaml(_CFG_SAME_ZONE_AND_CIDR)
+        spec.ConstructVirtualMachines()
+
+        self.assertDictContainsSubset({'cidr': '10.0.1.0/24'}, spec.custom_subnets['vm_1'])
+        self.assertDictContainsSubset({'cidr': '10.0.1.0/24'}, spec.custom_subnets['vm_2'])
+        self.assertLen(spec.networks, 1)
+        for k in spec.networks.keys():
+            self.assertItemsEqual(['10.0.1.0/24'], spec.networks[k].all_nets)
+
+    def testLoadSameZoneDiffCidrConfig(self):
+        spec = self._CreateBenchmarkSpecFromYaml(_CFG_SAME_ZONE_DIFF_CIDR)
+        spec.ConstructVirtualMachines()
+
+        self.assertDictContainsSubset({'cidr': '10.0.1.0/24'}, spec.custom_subnets['vm_1'])
+        self.assertDictContainsSubset({'cidr': '10.0.2.0/24'}, spec.custom_subnets['vm_2'])
+        self.assertLen(spec.networks, 2)
+        for k in spec.networks.keys():
+            self.assertItemsEqual(['10.0.1.0/24', '10.0.2.0/24'], spec.networks[k].all_nets)
+
+
+class TestGceNetworkNames(BaseGceNetworkTest):
+
+    def setUp(self):
+        super(TestGceNetworkNames, self).setUp()
+        # need a benchmarkspec in the context to run
+        FLAGS.run_uri = _URI
+        config_spec = benchmark_config_spec.BenchmarkConfigSpec(
+            'cluster_boot', flag_values=FLAGS)
+        benchmark_spec.BenchmarkSpec(mock.Mock(), config_spec, 'uid')
+
+    ########
+    # Network Names
+    ########
+    def testGetDefaultNetworkName(self):
+        project = _PROJECT
+        zone = 'us-north1-b'
+        cidr = None
+        # long_cidr = '123.567.901/13'  # @TODO net_utils for address sanity checks
+        vm = mock.Mock(zone=zone, project=project, cidr=cidr)
+        net = gce_network.GceNetwork.GetNetwork(vm)
+        net_name = net._MakeGceNetworkName()
+
+        _net_type = 'default'
+        _cidr_string = None
+        _uri = _URI
+        expected_netname = '-'.join(
+            i for i in ('pkb-network', _net_type, _cidr_string, _uri) if i and i not in ('default'))
+
+        self.assertEqual(expected_netname, net_name)  # pkb-network-uri45678 (default)
+        self.assertRegexpMatches(net_name, _REGEX_GCE_NET_NAMES)
+
+    def testGetSingleNetworkName(self):
+        FLAGS.gce_subnet_region = 'us-south1-c'
+        FLAGS.gce_subnet_addr = '2.2.3.4/33'
+
+        project = _PROJECT
+        zone = 'us-north1-b'
+        cidr = None
+        vm = mock.Mock(zone=zone, project=project, cidr=cidr)
+        net = gce_network.GceNetwork.GetNetwork(vm)
+        net_name = net._MakeGceNetworkName()
+
+        _net_type = 'single'
```

Comment:
I'm using underscore prefix for any variable contributing to the expected value in the test, as opposed to actual values returned from the method under test. It makes it easier (for me at least) to see where these values are coming from if a test fails since the variable names overlap. I'm using this convention for all the naming tests.

 Can you clarify this request? Would you like these variables removed and just hard code the expected_value string? Or just renamed to something else?